transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebSession.java

```diff
@@ -393,12 +421,16 @@ public void attachReader(final HttpSession newReader) {
             if (ALIGN_UPSTREAM) {
                 final Thread ioThread = getIoThread();
                 final Executor ioExecutor = getIoExecutor();
+                // Prevent messageReceived from being fired from setIoAlignment and racing ahead of attachReader0
+                newReader.suspendRead();
```

Comment:
I think we should backport this change (suspendRead/resumeRead) to 4.x.

I see the following in 4.x sometimes:
```
java.lang.IllegalStateException: not executable
build	17-Feb-2016 01:09:28	[INFO] 	at com.kaazing.mina.core.session.IoSessionEx$4.execute(IoSessionEx.java:65)
build	17-Feb-2016 01:09:28	[INFO] 	at com.kaazing.mina.core.filterchain.DefaultIoFilterChainEx.execute(DefaultIoFilterChainEx.java:197)
build	17-Feb-2016 01:09:28	[INFO] 	at com.kaazing.mina.core.filterchain.DefaultIoFilterChainEx.callNextExceptionCaught(DefaultIoFilterChainEx.java:148)
build	17-Feb-2016 01:09:28	[INFO] 	at com.kaazing.mina.core.filterchain.DefaultIoFilterChain.fireExceptionCaught(DefaultIoFilterChain.java:470)
build	17-Feb-2016 01:09:28	[INFO] 	at com.kaazing.mina.netty.IoSessionChannelHandler.exceptionCaught(IoSessionChannelHandler.java:63)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.Channels.fireExceptionCaught(Channels.java:525)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.AbstractChannelSink.exceptionCaught(AbstractChannelSink.java:48)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.Channels.fireChannelClosed(Channels.java:468)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.close(AbstractNioWorker.java:388)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:137)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:131)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:107)
build	17-Feb-2016 01:09:28	[INFO] 	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:232)
build	17-Feb-2016 01:09:28	[INFO] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1110)
build	17-Feb-2016 01:09:28	[INFO] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:603)
build	17-Feb-2016 01:09:28	[INFO] 	at java.lang.Thread.run(Thread.java:722)
```
Perhaps the above exception means : a close happened after the session migrated to NO_EXECUTOR and before it migrated to i/o executor.